plugins/onepad/Linux/opPanel.cpp

```diff
@@ -0,0 +1,220 @@
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ */
+
+#include "opPanel.h"
+#include "ImgHeader/dualshock2.h"
+#include "ImgHeader/cross.h"
+#include "ImgHeader/circle.h"
+#include "ImgHeader/square.h"
+#include "ImgHeader/triangle.h"
+#include "ImgHeader/dp_left.h"
+#include "ImgHeader/dp_right.h"
+#include "ImgHeader/dp_up.h"
+#include "ImgHeader/dp_bottom.h"
+#include "ImgHeader/l1.h"
+#include "ImgHeader/r1.h"
+#include "ImgHeader/l2.h"
+#include "ImgHeader/l3.h"
+#include "ImgHeader/r2.h"
+#include "ImgHeader/r3.h"
+#include "ImgHeader/start.h"
+#include "ImgHeader/select.h"
+#include "ImgHeader/analog.h"
+#include "ImgHeader/joystick_cursor.h"
+
+opPanel::opPanel(wxWindow *parent,
+                 wxWindowID id=wxID_ANY,
+                 const wxPoint &pos=wxDefaultPosition,
+                 const wxSize &size=wxDefaultSize
+                ): wxPanel( parent, id, pos, size)
+{
+    m_picture[img_background] = EmbeddedImage<res_dualshock2>().GetIcon();
+
+    m_picture[img_start] = EmbeddedImage<res_start>().GetIcon();
+    m_picture[img_select] = EmbeddedImage<res_select>().GetIcon();
+    m_picture[img_analog] = EmbeddedImage<res_analog>().GetIcon();
+
+    m_picture[img_dp_left] = EmbeddedImage<res_dp_left>().GetIcon();
+    m_picture[img_dp_right] = EmbeddedImage<res_dp_right>().GetIcon();
+    m_picture[img_dp_up] = EmbeddedImage<res_dp_up>().GetIcon();
+    m_picture[img_dp_bottom] = EmbeddedImage<res_dp_bottom>().GetIcon();
+
+    m_picture[img_square] = EmbeddedImage<res_square>().GetIcon();
+    m_picture[img_circle] = EmbeddedImage<res_circle>().GetIcon();
+    m_picture[img_cross] = EmbeddedImage<res_cross>().GetIcon();
+    m_picture[img_triangle] = EmbeddedImage<res_triangle>().GetIcon();
+
+    m_picture[img_l1] = EmbeddedImage<res_l1>().GetIcon();
+    m_picture[img_l3] = EmbeddedImage<res_l3>().GetIcon();
+    m_picture[img_l2] = EmbeddedImage<res_l2>().GetIcon();
+
+    m_picture[img_r1] = EmbeddedImage<res_r1>().GetIcon();
+    m_picture[img_r3] = EmbeddedImage<res_r3>().GetIcon();
+    m_picture[img_r2] = EmbeddedImage<res_r2>().GetIcon();
+
+    m_picture[img_left_cursor] = EmbeddedImage<res_joystick_cursor>().GetIcon();
+    m_picture[img_right_cursor] = EmbeddedImage<res_joystick_cursor>().GetIcon();
+
+    for(int i=0; i<NB_IMG; ++i)
+    {
+        m_refresh_required[i] = false;
+        SaveSize(i);
+        HideImg(i);
+    }
+    ShowImg(img_background);
+    m_refresh_required[img_background] = true;
+
+    m_left_cursor_x = 0;
+    m_left_cursor_y = 0;
+    m_right_cursor_x = 0;
+    m_right_cursor_y = 0;
+}
+
+void opPanel::SaveSize(int id)
+{
+    m_img_size[id][0] = m_picture[id].GetWidth();
+    m_img_size[id][1] = m_picture[id].GetHeight();
+}
+
+void opPanel::HideImg(int id)
+{
+    if(id < NB_IMG)
+    {
+        m_picture[id].SetWidth(0);
+        m_picture[id].SetHeight(0);
```

Comment:
Setting the width and height to 0 causes major issues with GTK3 - The visual button feedback doesn't work, and I get lots of warning spam.

It works after removing the lines, but if I press and release any button on my DS3, the D-pad left is shown as pressed when it is not.